### ISSUE TITLE
XML output in distributed (downloadable) documentation

### ISSUE THREAD

**Issue by albert-github** [CGAL/cgal#3770]
## Issue Details
In the release documentation set (e.g. https://www.cgal.org/2019/03/13/cgal414-beta2/ but also older versions) I saw that besides HTML also XML files are present.
- what is the purpose of these?
- are they tested?

When installing (on Windows) there was a naming conflict in the directory `doc_html/Circulator/xml` as there are the files (note the upper and lowercase `c`):
- `circulator_8h.xml`
- `Circulator_8h.xml`

When building on Windows I could not reproduce it as the files would overwrite each other (only `Circulator_8h.xml` is present but with a small `c`).

Are these files all recently generated (couldn't get it directly from the tar file as it is unclear whether or not the original dates are still there and not the copied dates)?
(in the HTML version neither `circulator_8h.html` nor `Circulator_8h.html` is present due to a different structure of the output.

**Comment by sloriot** [comment]
@maxGimeno I looks like a side effect of your script in the testsuite.

**Comment by albert-github** [comment]
@sloriot, @maxGimeno 
Even if it is only a result in the test suite I think it is god to know about as in that case doxygen, probably, generates 2 files with names that cannot really be distinguished (especially on Windows type machines, including Cygwin) and we should get to the cause of of so that either doxygen can provide a fix or give an advise on how to fix it.

**Comment by maxGimeno** [comment]
The xml files are used in the process of testing the doc. They are parsed to procuce the [doxygen testpage](https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-4.14-Ic-147/) in our testsuite. They should only be produced when testing the doc in the testsuite and not be present in the distributed doc IMO, as i don't think they are used in the post-processing. I'd say the simpler fix would be to generate the doc of the release without the tag for XML, and not only take the last good testsuite.

**Comment by albert-github** [comment]
I think that one could take the last good testsuite and exclude the `xml` directory when packaging

**Comment by maxGimeno** [comment]
Yes, I guess that should work too.

**Comment by albert-github** [comment]
I think I found the reason for the `circulator_8h.xml` / `Circulator_8h.xml` problem.
When looking in the `Circulator` source directory fo `.h` files we see:
```
./doc/Circulator/CGAL/circulator.h
./doc/Circulator/CGAL/circulator_bases.h
./doc/Circulator/Concepts/Circulator.h
./doc/Circulator/Concepts/ConstHandle.h
./doc/Circulator/Concepts/ConstRange.h
./doc/Circulator/Concepts/Handle.h
./doc/Circulator/Concepts/Range.h
./include/CGAL/Circulator/Circulator_adapters.h
./include/CGAL/Circulator/Circulator_concepts.h
./include/CGAL/Circulator/Safe_circulator_from_iterator.h
./include/CGAL/circulator.h
./include/CGAL/circulator_bases.h
```

so in the `doc` sub-directory we have:
```
./doc/Circulator/CGAL/circulator.h
./doc/Circulator/Concepts/Circulator.h
```
maybe the `./doc/Circulator/Concepts/Circulator.h` should be renamed to `./doc/Circulator/Concepts/Circulator_Concepts.h` to (also) conform to the name in the `include` sub-directory.
The advantages:
- correct xml documentation
- avoid name clash on Windows (probably not CGAL's target but in future ...) and other case insensitive type of systems 
- less confusing

**Comment by albert-github** [comment]
Maybe worth an new issue, but let me pose the question here:
are there any intentions to correct the errors as shown by running doxygen withe `GENERATE_XML=YES`?
at the moment I get 47 lines of output, but just in 3 files:
```
Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h
Poisson_surface_reconstruction_3/include/CGAL/Poisson_reconstruction_function.h
Poisson_surface_reconstruction_3/include/CGAL/Reconstruction_triangulation_3.h
Poisson_surface_reconstruction_3/include/CGAL/IO/output_surface_facets_to_triangle_soup.h
BGL/doc/BGL/Concepts/FaceGraph.h
BGL/doc/BGL/Concepts/HalfedgeGraph.h
/cygdrive/d/Programs/CGAL/cgal_fork/AABB_tree/include/CGAL/internal/AABB_tree/AABB_node.h
```
The warnings are a result of the fact that XML, automatically, also outputs the private entities (it is up to the further processing software to ignore the private entities. In this way the further processing is the most flexbile and has he most possibilities).

**Comment by albert-github** [comment]
Made a proposed pull request in doxygen (https://github.com/doxygen/doxygen/pull/6890) with some corrections based on (a.o.) the CGAL XML output

**Comment by sloriot** [comment]
XML output is used to check that there is no regression (exactly the same things are documented). If this is something easy to do, we will if not I guess we won't.

**Comment by lrineau** [comment]
I have fixed that issue in commit fc18d4c3da4.
